File: extract_schema.js
const fs = require('fs');
const content = fs.readFileSync('c:/Users/Hon3y Chauhan/Desktop/NAVTTC-LMS/index.html', 'utf-8');

const tables = {};
function addCol(table, col) {
    if (!tables[table]) tables[table] = new Set();
    if (col && col !== '*') tables[table].add(col);
}

// match .from('table') followed by anything until next .from or end
const tableRegex = /\.from\(['"]([^'"]+)['"]\)([\s\S]*?)(?=\.from|\Z)/g;
let match;
while ((match = tableRegex.exec(content)) !== null) {
    const table = match[1];
    const ops = match[2];

    const selectRegex = /\.select\(['"]([^'"]+)['"]\)/g;
    let sm;
    while ((sm = selectRegex.exec(ops)) !== null) {
        sm[1].split(',').forEach(c => addCol(table, c.trim()));
    }

    const eqRegex = /\.eq\(['"]([^'"]+)['"]/g;
    let em;
    while ((em = eqRegex.exec(ops)) !== null) {
        addCol(table, em[1]);
    }

    const payloadRegex = /\.(?:insert|update|upsert)\(\s*({[^}]+})\s*\)/g;
    let pm;
    while ((pm = payloadRegex.exec(ops)) !== null) {
        const payload = pm[1];
        const keyRegex = /['"]?([a-zA-Z0-9_]+)['"]?\s*:/g;
        let km;
        while ((km = keyRegex.exec(payload)) !== null) {
            addCol(table, km[1]);
        }
    }
}

try {
    const content2 = fs.readFileSync('c:/Users/Hon3y Chauhan/Desktop/NAVTTC-LMS/verify.html', 'utf-8');
    let match2;
    while ((match2 = tableRegex.exec(content2)) !== null) {
        const table = match2[1];
        const ops = match2[2];
        const selectRegex = /\.select\(['"]([^'"]+)['"]\)/g;
        let sm;
        while ((sm = selectRegex.exec(ops)) !== null) {
            sm[1].split(',').forEach(c => addCol(table, c.trim()));
        }
        const eqRegex = /\.eq\(['"]([^'"]+)['"]/g;
        let em;
        while ((em = eqRegex.exec(ops)) !== null) {
            addCol(table, em[1]);
        }
    }
} catch (e) {}

const res = {};
for (const t in tables) {
    res[t] = Array.from(tables[t]);
}
console.log(JSON.stringify(res, null, 2));

File: extract_schema.py
import re
import json
from collections import defaultdict

with open('c:/Users/Hon3y Chauhan/Desktop/NAVTTC-LMS/index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Pattern to find client.from('table').insert({ ... }) or upsert or update
# This is a bit complex for a simple regex, but we can look for basic patterns
# like client.from('table').select('col1, col2')

tables = defaultdict(set)

# Find all .from('table')
table_matches = re.finditer(r"\.from\(['\"]([^'\"]+)['\"]\)(.*?)(?=\.from|\Z)", content, re.DOTALL)

for match in table_matches:
    table_name = match.group(1)
    operations = match.group(2)
    
    # look for selects
    selects = re.findall(r"\.select\(['\"]([^'\"]+)['\"]\)", operations)
    for s in selects:
        cols = [c.strip() for c in s.split(',')]
        for c in cols:
            if c != '*' and c != '':
                tables[table_name].add(c)
                
    # look for eq
    eqs = re.findall(r"\.eq\(['\"]([^'\"]+)['\"]", operations)
    for eq in eqs:
        tables[table_name].add(eq)
        
    # try to find insert/update/upsert payloads
    # This might match the immediate next object
    payloads = re.findall(r"\.(?:insert|update|upsert)\(\s*({[^}]+})\s*\)", operations)
    for p in payloads:
        # extract keys from the JSON-like object
        keys = re.findall(r"['\"]?([a-zA-Z0-9_]+)['\"]?\s*:", p)
        for k in keys:
            tables[table_name].add(k)

# Additional file
try:
    with open('c:/Users/Hon3y Chauhan/Desktop/NAVTTC-LMS/verify.html', 'r', encoding='utf-8') as f:
        content2 = f.read()
    
    table_matches2 = re.finditer(r"\.from\(['\"]([^'\"]+)['\"]\)(.*?)(?=\.from|\Z)", content2, re.DOTALL)
    for match in table_matches2:
        table_name = match.group(1)
        operations = match.group(2)
        selects = re.findall(r"\.select\(['\"]([^'\"]+)['\"]\)", operations)
        for s in selects:
            cols = [c.strip() for c in s.split(',')]
            for c in cols:
                if c != '*' and c != '':
                    tables[table_name].add(c)
        eqs = re.findall(r"\.eq\(['\"]([^'\"]+)['\"]", operations)
        for eq in eqs:
            tables[table_name].add(eq)
except Exception as e:
    pass

for t in tables:
    tables[t] = list(tables[t])

print(json.dumps(tables, indent=2))
